modules/docs/arrow-docs/docs/docs/arrow/typeclasses/traverse/README.md

```diff
@@ -11,16 +11,366 @@ redirect_from:
+
+```kotlin:ank:playground
+import arrow.core.Option
+import arrow.core.extensions.list.traverse.sequence
+import arrow.core.extensions.option.applicative.applicative
+import arrow.core.none
+import arrow.core.some
+
+fun main() {
+  //sampleStart
+  val optionList =
+    listOf(1.some(), 2.some(), 3.some())
+      .sequence(Option.applicative())
+
+  val emptyList =
+    listOf(1.some(), none(), 3.some())
+      .sequence(Option.applicative())
+  //sampleEnd
+  println("optionList = $optionList")
+  println("emptyList = $emptyList")
+}
+```
+
+-- Add Text
+
+### Traversables are Functors
+
+{TODO: finish this section}
+
+### Traversing for effect
+
+Sometimes our effectful functions return a `Unit` value in cases where there is no interesting value to return (e.g. writing to some sort of store).
+
+```kotlin:ank
+import arrow.fx.ForIO
+import arrow.fx.Promise
+
+interface Data
+
+fun writeToStore(data:Data): Promise<ForIO, Unit> = TODO()
+```
+
+If we traverse using this, we end up with a funny type.
+
+```kotlin:ank
+import arrow.core.ListK
+import arrow.fx.IO
+import arrow.fx.extensions.io.applicative.applicative
+import arrow.fx.fix
+
+fun writeManyToStore(data: ListK<Data>): IO<ListK<Unit>> =
+  data.traverse(IO.applicative()) { writeToStore(it).get().fix() }.fix()
+```
+
+We end up with a `IO<ListK<Unit>>`! A `ListK<Unit>` is not of any use to us, and communicates the same amount of information as a single `Unit` does.
+
+Traversing solely for the sake of the effect (ignoring any values that may be produced, `Unit` or otherwise) is common, so `Foldable` (superclass of `Traverse`) provides `traverse_` and `sequence_` methods that do the same thing as `traverse` and `sequence` but ignores any value produced along the way, returning `Unit` at the end.
+
+```kotlin:ank
+fun writeManyToStore(data: ListK<Data>): IO<Unit> =
+  data.traverse_(IO.applicative()) { writeToStore(it).get() }.fix()
+```
+```kotlin:ank:playground
+import arrow.core.Option
+import arrow.core.extensions.list.foldable.sequence_
+import arrow.core.extensions.option.applicative.applicative
+import arrow.core.none
+import arrow.core.some
+
+fun main() {
+  //sampleStart
+  val someUnit = listOf(1.some(), 2.some(), 3.some())
+    .sequence_(Option.applicative())
+  val noneUnit = listOf(1.some(), none(), 3.some())
+    .sequence_(Option.applicative())
+  //sampleEnd
+  println("someUnit= $someUnit")
+  println("noneUnit= $noneUnit")
+}
+```
+
+### Traversables are Foldable
+
+The `Foldable` type class abstracts over “things that can be folded over” similar to how `Traverse` abstracts over “things that can be traversed.” It turns out `Traverse` is strictly more powerful than `Foldable` - that is, `foldLeft` and `foldRight` can be implemented in terms of `traverse` by picking the right `Applicative`. However, arrow's `Traverse` does not implement `foldLeft` and `foldRight` as the actual implementation tends to be ineffecient, as we one can implement the `Foldable` `fold`'s.
```

Comment:
This is just a guess, but I am guessing that the Foldable fold is more efficient, because traverse can blowUp the stack if it isn't defined by fold. 
https://github.com/typelevel/cats/issues/1473